File: algorithm/typescript/BOJ10809.ts
// problem : https://www.acmicpc.net/problem/10809

import fs = require("fs");

const inputString: string = fs.readFileSync("/dev/stdin").toString();
const alphabet: number[] = new Array(26).fill(-1);

for (let i = 0, size = inputString.length; i < size; i++) {
  if (alphabet[inputString.charCodeAt(i) - 97] === -1) {
    alphabet[inputString.charCodeAt(i) - 97] = i;
  }
}

console.log(alphabet.join(" "));

File: algorithm/python/BOJ10809.py
# problem : https://www.acmicpc.net/problem/10809

import sys

string = sys.stdin.readline().rstrip()
alphabet = [-1] * 26

for i in range(0, len(string)):
    if alphabet[ord(string[i]) - ord("a")] == -1:
        alphabet[ord(string[i]) - ord("a")] = i

for char in alphabet:
    print(char, end=" ")
